packages/react/src/components/FileUploader/stories/drop-container.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useCallback } from 'react';
+import { settings } from 'carbon-components';
+import fetchMock from 'fetch-mock/src/client.js';
+import FileUploaderItem from '../FileUploaderItem';
+import FileUploaderDropContainer from '../FileUploaderDropContainer';
+import FormItem from '../../FormItem';
+import uid from '../../../tools/uniqueId';
+
+const { prefix } = settings;
+
+fetchMock.mock({
+  method: 'POST',
+  matcher: '*',
+  overwriteRoutes: false,
+  response: () =>
+    new Promise((resolve, reject) =>
+      setTimeout(
+        () => (Math.round(Math.random()) ? resolve('200') : reject(503)),
+        Math.random() * 1000
+      )
+    ),
+});
+
+function ExampleDropContainerApp(props) {
+  const [files, setFiles] = useState([]);
+  const uploadFile = async fileToUpload => {
+    try {
+      await fetch('https://ibm.com/', {
+        method: 'POST',
+        body: fileToUpload,
+      });
+      const updatedFile = {
+        ...fileToUpload,
+        status: 'complete',
+        iconDescription: 'Upload complete',
+      };
+      setFiles(files =>
+        files.map(file =>
+          file.uuid === fileToUpload.uuid ? updatedFile : file
+        )
+      );
+      setTimeout(() => {
+        const updatedFile = {
+          ...fileToUpload,
+          status: 'edit',
+          iconDescription: 'Delete file',
+        };
+        setFiles(files =>
+          files.map(file =>
+            file.uuid === fileToUpload.uuid ? updatedFile : file
+          )
+        );
+      }, 1000);
+    } catch (error) {
+      const updatedFile = {
+        ...fileToUpload,
+        status: 'edit',
+        iconDescription: 'Upload failed',
+        invalid: true,
+      };
+      setFiles(files =>
+        files.map(file => (file === fileToUpload ? updatedFile : file))
+      );
+      console.log(error); // eslint-disable-line no-console
+    }
+  };
+  const onAddFiles = useCallback(
+    (evt, { addedFiles }) => {
+      evt.stopPropagation();
+      const newFiles = addedFiles.map(file => ({
+        uuid: uid(),
+        name: file.name,
+        size: file.size,
+        status: 'uploading',
+        iconDescription: 'Uploading',
+      }));
+      props.multiple
+        ? setFiles([...files, ...newFiles])
+        : setFiles([...files, newFiles[0]]);
+      newFiles.forEach(uploadFile);
+    },
+    [files, props.multiple]
+  );
+  const handleFileUploaderItemClick = useCallback(
+    (evt, { uuid: clickedUuid }) =>
+      setFiles(files.filter(({ uuid }) => clickedUuid !== uuid)),
+    [files]
+  );
+  return (
+    <FormItem>
+      <strong className={`${prefix}--file--label`}>Account photo</strong>
+      <p className={`${prefix}--label-description`}>
+        Only .jpg and .png files. 500kb max file size.
+      </p>
+      <FileUploaderDropContainer {...props} onAddFiles={onAddFiles}>
+        {files.map(
```

Comment:
My apologies for being late to notice this; Does `<FileUploaderItem>` in `<FileUploaderDropContainer>` mean we are doing drop effect in the "file name list" area...? I'm asking as I was imagining `<FileUploaderDropContainer>` not containing `<FileUploaderItem>`, but I may be wrong.

Reply:
instead of making part of the label interactive, would it be better to make the entire drop container interactive?

so the component would look like this on hover or focus, and we would be able to more easily support translation

![image](https://user-images.githubusercontent.com/8265238/60518173-9633eb80-9ca6-11e9-8d2a-534a422d91a9.png)

`<FileUploaderItem>` doesn't need to be a child of the drop container, I think either approach is fine but I'll edit the example

Reply:
going to mark this resolved since the first point seems to be duplicate of https://github.com/carbon-design-system/carbon/pull/3175#discussion_r299323685